go/store/val/codec.go

```diff
@@ -430,14 +430,29 @@ func compareDecimal(l, r decimal.Decimal) int {
 }
 
 const minYear int16 = 1901
+const maxYear int16 = 2155
+const zeroToken uint8 = 255
```

Comment:
how would you feel about:
```go
type yearOffset uint8
const (
        maxOffset yearOffset = 254
        nilOffset = 255
)
```
Not sure it's strictly better, but it seems like we are encoding offsets? And this might give `zeroToken` a more sensible name in that context.